lib/src/layer/tile_layer/tile_provider/network_image_provider.dart

```diff
@@ -1,73 +1,119 @@
+  /// The headers to include with the tile fetch request
   final Map<String, String> headers;
 
-  FMNetworkImageProvider(
-    this.url, {
+  /// Dedicated [ImageProvider] to fetch tiles from the network
+  FlutterMapNetworkImageProvider({
+    required this.url,
     required this.fallbackUrl,
-    http.Client? httpClient,
-    this.headers = const {},
-  }) : httpClient = httpClient ?? RetryClient(http.Client());
+    required this.headers,
+    required this.httpClient,
+  });
+
+  // TODO: These [load] and [loadBuffer] method ensure support for Flutter 3.3
+  // thru 3.10, hence the multiple deprecation ignorances. Once the methods &
+  // types have been removed, they need to be replaced with [loadImage], and the
+  // min SDK constraint will need to be bumped.
 
   @override
-  ImageStreamCompleter loadBuffer(
-      FMNetworkImageProvider key, DecoderBufferCallback decode) {
-    return OneFrameImageStreamCompleter(_loadWithRetry(key, decode),
-        informationCollector: () sync* {
-      yield ErrorDescription('Image provider: $this');
-      yield ErrorDescription('Image key: $key');
-    });
+  ImageStreamCompleter load(
+    FlutterMapNetworkImageProvider key,
+    // ignore: deprecated_member_use
+    DecoderCallback decode,
+  ) {
+    final StreamController<ImageChunkEvent> chunkEvents =
+        StreamController<ImageChunkEvent>();
+
+    return MultiFrameImageStreamCompleter(
+      codec: _loadAsync(key, chunkEvents, decodeDepreacted: decode),
+      chunkEvents: chunkEvents.stream,
+      scale: 1,
+      debugLabel: key.url,
+      informationCollector: () => <DiagnosticsNode>[
+        DiagnosticsProperty('URL', url),
+        DiagnosticsProperty('Fallback URL', fallbackUrl),
+        DiagnosticsProperty('Current provider', key),
+      ],
+    );
   }
 
   @override
-  Future<FMNetworkImageProvider> obtainKey(ImageConfiguration configuration) {
-    return SynchronousFuture<FMNetworkImageProvider>(this);
+  ImageStreamCompleter loadBuffer(
+    FlutterMapNetworkImageProvider key,
+    // ignore: deprecated_member_use
+    DecoderBufferCallback decode,
+  ) {
+    final StreamController<ImageChunkEvent> chunkEvents =
+        StreamController<ImageChunkEvent>();
+
+    return MultiFrameImageStreamCompleter(
+      codec: _loadAsync(key, chunkEvents, decode: decode),
+      chunkEvents: chunkEvents.stream,
+      scale: 1,
+      debugLabel: key.url,
+      informationCollector: () => <DiagnosticsNode>[
+        DiagnosticsProperty('URL', url),
+        DiagnosticsProperty('Fallback URL', fallbackUrl),
+        DiagnosticsProperty('Current provider', key),
+      ],
+    );
   }
 
-  Future<ImageInfo> _loadWithRetry(
-    FMNetworkImageProvider key,
-    DecoderBufferCallback decode, [
+  Future<Codec> _loadAsync(
+    FlutterMapNetworkImageProvider key,
+    StreamController<ImageChunkEvent> chunkEvents, {
+    // ignore: deprecated_member_use
+    DecoderBufferCallback? decode,
+    // ignore: deprecated_member_use
```

Comment:
This isn't necessarily for removal in v5. It's for removal whenever we want to stop supporting Flutter <3.10.
This update has breaking changes anyway, so it would need to be v5.

Reply:
Imo we should support v3.10 in the next major version, there's no point to maintain older Flutter version 🤔

Reply:
So you want to remove support for everything below 3.10 in the next version?